### PR TITLE
Everviz documentation start

### LINKED ISSUE #40: Add documentation
Add sphinx-based documenation

### PR DESCRIPTION
**Pull request by verveerpj**
Initial documentation for everviz

closes #40

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# This pattern also affects html_static_path and html_extra_path.
+exclude_patterns = []
+
+
+# -- Options for HTML output -------------------------------------------------
+
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+#
+html_theme = 'classic'
```
Comment (oyvindeide) [inline doc/conf.py:57]: Can I suggest a more modern theme:
```suggestion
html_theme = "sphinx_rtd_theme"
```

File: doc/conf.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+release = version
+
+
+# -- General configuration ---------------------------------------------------
+
+# Add any Sphinx extension module names here, as strings. They can be
+# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
+# ones.
+extensions = [
+]
```
Comment (oyvindeide) [inline doc/conf.py:41]:
```suggestion
extensions = [
    "sphinx_rtd_theme",
]
```

File: doc/conf.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+# https://www.sphinx-doc.org/en/master/usage/configuration.html
+
+# -- Path setup --------------------------------------------------------------
+
+# If extensions (or modules to document with autodoc) are in another directory,
+# add these directories to sys.path here. If the directory is relative to the
+# documentation root, use os.path.abspath to make it absolute, like shown here.
+
+import pkg_resources
+
```
Comment (oyvindeide) [inline doc/conf.py:14]:
```suggestion
import sphinx_rtd_theme
```

**Comment by verveerpj** [comment]
re suggestions @oyvindeide :
- new theme
- renamed doc to docs, inline with what everest does
- I have not included the proposed changes to tox.ini since I cannot judge/test the changes yet. Shall we do that in a separate step?

**Comment by oyvindeide** [comment]
>     * I have not included the proposed changes to tox.ini since I cannot judge/test the changes yet. Shall we do that in a separate step?

Not sure I understand what you mean? Adding this to `tox.ini` and running `tox -e docs` locally on your branch builds the docs and I can inspect them:

```
[testenv:docs]
description = invoke sphinx-build to build the HTML docs
deps =
    sphinx
    sphinx-rtd-theme
commands = sphinx-build -W -b html -d {envtmpdir}/doctrees docs {envtmpdir}/html
```

Then we can add that command to `.travis.yml` and it should run there as well.

**Comment by verveerpj** [comment]
> Not sure I understand what you mean? Adding this to `tox.ini` and running `tox -e docs` locally on your branch builds the docs and I can inspect them:

ah, now I see, never used tox, but just a pip install tox did the trick to test it... I have added it.

**Pull request merged**